### PR TITLE
Update XCode to fix iOS deploy

### PR DESCRIPTION
**Pull request by Jag96**
cc @roryabraham 

### Details
This PR updates the XCode version to fix an issue where fastlane is failing because of a mismatch.

### Fixed Issues

$ https://github.com/Expensify/App/issues/4337

### Tests
Building the app and archiving it works fine for me, so the next best way to test this is to merge and confirm the deploy doesn't fail

### QA Steps
None

### Tested On
- [x] iOS

### Screenshots
N/A

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -564,7 +564,7 @@ PODS:
     - UMCore
   - UMConstantsInterface (6.1.0):
     - UMCore
-  - UMCore (7.1.0)
+  - UMCore (7.1.1)
```
Comment (Jag96) [inline ios/Podfile.lock:567]: These automatically updated when running `pod install`

**Review by AndrewGable — DISMISSED** [review]

**Comment by Jag96** [comment]
Looks like e2e fails now, will likely have to fix broken dependencies 👀

**Comment by Jag96** [comment]
From the failing build, looks like `/Applications/Xcode_12.5.app/` doesn't have anything there
```
Fetching aws-sdk-kms 1.43.0
Installing aws-sdk-kms 1.43.0
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
/Users/runner/work/App/App/vendor/bundle/ruby/2.7.0/gems/json-2.5.1/ext/json/ext/generator
/Users/runner/hostedtoolcache/Ruby/2.7.4/x64/bin/ruby -I
/Users/runner/hostedtoolcache/Ruby/2.7.4/x64/lib/ruby/2.7.0 -r
./siteconf20210730-1267-rn4l0w.rb extconf.rb
creating Makefile

current directory:
/Users/runner/work/App/App/vendor/bundle/ruby/2.7.0/gems/json-2.5.1/ext/json/ext/generator
make "DESTDIR=" clean
xcrun: error: missing DEVELOPER_DIR path:
/Applications/Xcode_12.5.app/Contents/Developer

current directory:
/Users/runner/work/App/App/vendor/bundle/ruby/2.7.0/gems/json-2.5.1/ext/json/ext/generator
make "DESTDIR="
xcrun: error: missing DEVELOPER_DIR path:
/Applications/Xcode_12.5.app/Contents/Developer

make failed, exit code 1
```

**Comment by Jag96** [comment]
Looks like GH actions available versions only go up to 12.4 for `macos-latest` which is 10.15:
- https://github.com/actions/virtual-environments#available-environments
- https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#xcode

So for 12.5 we'd have to update to `macos-11` per https://github.com/actions/virtual-environments#available-environments
- https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11-Readme.md#xcode

Going to give that a shot and see what the result is

**Comment by Jag96** [comment]
Looks like we need to match the schemas in https://json.schemastore.org/github-workflow.json, so `macos-11.0` instead of `macos-11`. However, that seems to break the e2e (which didn't break with `macos-11`). I believe we'll have to update the `validateActionsAndWorkflows` script to fix this cc @roryabraham.

image

**Comment by aldo-expensify** [comment]
Sorry @Jag96 , I don't know how to test this, and it will take me some time to figure it out. I'll be off on Monday, so you may want to choose someone with more experience to review this :)

**Comment by roryabraham** [comment]
Great investigation so far @Jag96!

> I believe we'll have to update the validateActionsAndWorkflows script to fix this cc @roryabraham.

Okay, I guess for now the best thing to do might be to:

1. Switch back to `runs-on: macos-11`
1. Just disable that check in our CI/CD for now
1. Submit a PR against schemastore to update the JSON schema.
1. Once that PR is merged, re-enable the schema verification.

**Comment by roryabraham** [comment]
SchemaStore PR [here](https://github.com/SchemaStore/schemastore/pull/1777)

**Comment by roryabraham** [comment]
Whoa, the SchemaStore PR is already merged, so the schema validation check is passing ✅ 🎉

**Comment by Jag96** [comment]
Putting this into review, looks like e2e got past the part where it was previously failing so this should be good 🤞

**Review by roryabraham — APPROVED** [review]
Okay, this seems pretty simple – update the runners to `macos-11` and xcode to `12.5`. Hopefully this works!

**Review by AndrewGable — APPROVED** [review]

**Pull request merged**